lib/util/cmcd_manager.js

```diff
@@ -349,6 +357,29 @@ shaka.util.CmcdManager = class {
     }
   }
 
+  /**
+   * Removes the CMCD query parameter from a URI.
+   *
+   * @param {string} uri
+   * @return {string}
+   * @private
+   */
+  removeCmcdQueryFromUri_(uri) {
+    if (!uri || !uri.includes('CMCD=')) {
```

Comment:
Any reason uri is optional? Shall we not call this func when no uri is provided at the caller?